enhancements/api-review/custom-tags-aws.md

```diff
@@ -75,10 +75,70 @@ will apply these tags to all AWS resources they create.
 
 userTags that are specified in the infrastructure resource will merge with userTags specified in an individual component. In the case where a userTag is specified in the infrastructure resource and there is a tag with the same key specified in an individual component, the value from the individual component will have precedence and be used.
 
-The userTags field is intended to be set at install time and is considered immutable. Components that respect this field must only ever add tags that they retrieve from this field to cloud resources, they must never remove tags from the existing underlying cloud resource even if the tags are removed from this field(despite it being immutable).
+Users can update the `resourceTags` by editing `.spec.aws` of the `infrastructure.config.openshift.io` type. New addition of tags are always appended. Any update in the existing tags, which are added by installer or  by edit in `resourceTags`, will replace the previous tag value.
```

Comment:
> Are we sure that all resources will be updated with the new tag values?

As I understand it, the point of this PR is to make it so.  

> Are there any that are create and forget?

This needs to be clarified in this PR.  The enhancement summary only lists one component (the ingress operator) that required changes for the original user-defined AWS tags API implementation.  However, there are probably other components that also required changes.  We'll need to make sure that every team that has a component that required changes be aware of this update to the API.  We really don't want to get into a state where updating tags is kind of sort of supported because some components handle updates while others do not.

Reply:
> Are we sure that all resources will be updated with the new tag values? Are there any that are create and forget?

@JoelSpeed EBS would be a candidate as we lose track of the volumes and won't attempt to re-tag existing ones. This should also be discussed here, I think we added this into our user story back then when we refined it.

@Miciah you can find some overview of all components that have tags and need them updated as stories in the epic: https://issues.redhat.com/browse/CFE-69

I agree that this should be included in here as a listing and we should get the respective teams to review too.

Reply:
How do we find out which components create other AWS resources, for example, I'm working on adding support for a new AWS resource, Placement Groups, and at the moment, we aren't tagging it with user defined tags. We will need to do so for GA but it's not in the MVP. 

This is another resource where we are creating but not updating the object and so I'm not sure how the tagging would work here, I guess it will be like the EBS situation?

Reply:
> How do we find out which components create other AWS resources, for example, I'm working on adding support for a new AWS resource, Placement Groups, and at the moment, we aren't tagging it with user defined tags. We will need to do so for GA but it's not in the MVP.

I've been looking into this a couple months back, one can tap into CloudTrail and see what resources were PUT by the IAM user that created the cluster and then request the resource tags by the respective resource ID (then check for missing tags etc). We have to setup a proper environment for this I believe, so we didn't pursue this any further because of the effort.

Do you know how many new AWS (taggable) resources we introduce into core OpenShift per release? I was under the impression that this will continuously reduce towards zero as we focus on addon and component optionality. `Placement Groups` is certainly a great counter-example :) 

> This is another resource where we are creating but not updating the object and so I'm not sure how the tagging would work here, I guess it will be like the EBS situation?

I guess so, and there will for sure be other cases that could also be outside of our control in the future. We need to ensure to document them properly.

Reply:
> I've been looking into this a couple months back, one can tap into CloudTrail and see what resources were PUT by the IAM user that created the cluster and then request the resource tags by the respective resource ID (then check for missing tags etc). We have to setup a proper environment for this I believe, so we didn't pursue this any further because of the effort.

This sounds like something we could possibly codify into an E2E test, have you considered that?

> Do you know how many new AWS (taggable) resources we introduce into core OpenShift per release? I was under the impression that this will continuously reduce towards zero as we focus on addon and component optionality. Placement Groups is certainly a great counter-example :)

Our team frequently adds support for new EC2 features to the Machine API. Sometimes these rely on creating additional resources, as with the Placement Group example. I doubt we will see that slow down as we still have many customer RFEs asking for support for various use cases on each of the clouds

> I guess so, and there will for sure be other cases that could also be outside of our control in the future. We need to ensure to document them properly.

Agreed, would be good to enumerate all of the components and the resources they create and have that documented. Sounds like your CloudTrail thing could be used to generate a list?

Reply:
> This sounds like something we could possibly codify into an E2E test, have you considered that?

I do remember that there was a problem with this approach and thus we dropped it, @TrilokGeer it's probably good to spin up a R&D task to figure out if we can test this case in the e2e too. @thejasn is working on testing right now I believe.

Agreed, if we can automate this to generate a list, that would be even cooler!

Reply:
Thanks for the query. Yes, there are resources that are create and forget, for example, resources created by csi drivers.

Reply:
@tjungblu @JoelSpeed yes, this can be checked out. Thanks.